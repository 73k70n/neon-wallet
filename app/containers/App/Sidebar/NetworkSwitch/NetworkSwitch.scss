@import '../../../../styles/variables';

.networkSwitch {
  color: #666;
  font-size: 0.8em;
  border-radius: 0px;
  width: 150px;

  .networkSwitchTextInputWrapper {
    margin-bottom: 0;
  }

  .networkSwitchTextInputContainer {
    background-color: transparent;
    border: none;
  }

  .networkSwitchActive {
    background-color: transparent;
    border: none;
  }

  .label {
    margin-right: 4px;
  }

  .dropdownItem {
    padding: 10px 20px;
    font-size: 16px;
    cursor: pointer;

    &:hover {
      background: #e6e6e6;
    }

    .highlight {
      font-weight: 600;
    }
  }
}
