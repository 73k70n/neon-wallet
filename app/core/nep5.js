// @flow
import { map } from 'lodash'

import { toBigNumber } from './math'
import { COIN_DECIMAL_LENGTH } from './formatters'
import { TOKENS, TOKENS_TEST, MAIN_NETWORK_ID, TEST_NETWORK_ID } from './constants'

let fetchedTokens;

export const adjustDecimalAmountForTokenTransfer = (value: string): string =>
  toBigNumber(value).times(10 ** COIN_DECIMAL_LENGTH).round().toNumber()

const getTokenEntry = ((): Function => {
  let id = 1

  return (symbol: string, scriptHash: string, networkId: string) => ({
    id: `${id++}`,
    symbol,
    scriptHash,
    networkId,
    isUserGenerated: false
  })
})()

export const getDefaultTokens = async (): Promise<Array<TokenItemType>> => {
  const tokens = []

  // Prevent duplicate requests here
  if (!fetchedTokens) {
    const response = await fetch('https://raw.githubusercontent.com/CityOfZion/neo-tokens/master/tokenList.json')
    fetchedTokens = await response.json()
  }

  tokens.push(...map(fetchedTokens, (tokenData) => getTokenEntry(tokenData.symbol,tokenData.networks['1'].hash, MAIN_NETWORK_ID)))
  tokens.push(...map(TOKENS_TEST, (scriptHash, symbol) => getTokenEntry(symbol, scriptHash, TEST_NETWORK_ID)))

  return tokens
}
