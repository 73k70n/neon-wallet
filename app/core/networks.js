// @flow
import { MAIN_NETWORK_ID, TEST_NETWORK_ID } from './constants'

<<<<<<< HEAD
export const isMainNetwork = (networkId: string) =>
  networkId === MAIN_NETWORK_ID
export const isTestNetwork = (networkId: string) =>
  networkId === TEST_NETWORK_ID
=======
export const isMainNetwork = (networkId: string): boolean => networkId === MAIN_NETWORK_ID
export const isTestNetwork = (networkId: string): boolean => networkId === TEST_NETWORK_ID
export const isCozNetwork = (networkId: string): boolean => networkId === COZ_TEST_NETWORK_ID
>>>>>>> be3d2cb... squashed release 0.2.7 into one commit

export const getNetworks = () => [
  {
    id: MAIN_NETWORK_ID,
    label: 'MainNet',
    network: 'MainNet'
  },
  {
    id: TEST_NETWORK_ID,
    label: 'TestNet',
    network: 'TestNet'
  }
]

export const findNetwork = (networkId: string): NetworkItemType => {
  const networks = getNetworks()
  return networks.find(({ id }) => id === networkId) || networks[0]
}
