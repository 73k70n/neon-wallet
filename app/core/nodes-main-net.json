[
  {
    "protocol": "https",
    "url": "seed1.switcheo.network",
    "location": "Singapore",
    "port": "10331",
    "locale": "sg",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed2.switcheo.network",
    "location": "Singapore",
    "port": "10331",
    "locale": "sg",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed3.switcheo.network",
    "location": "Singapore",
    "port": "10331",
    "locale": "sg",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed1.travala.com",
    "location": "Singapore",
    "address": "52.74.91.157",
    "locale": "sg",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed2.travala.com",
    "location": "Seoul",
    "address": "13.209.29.171",
    "locale": "kr",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed3.travala.com",
    "location": "Canada",
    "address": "35.182.220.56",
    "locale": "ca",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed4.travala.com",
    "location": "London",
    "address": "52.56.132.44",
    "locale": "gb",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed1.neo.org",
    "location": "China",
    "address": "47.94.164.49",
    "locale": "cn",
    "port": "10331",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed2.neo.org",
    "location": "China",
    "address": "139.219.106.33",
    "locale": "cn",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed3.neo.org",
    "location": "Canada",
    "address": "47.89.178.72",
    "locale": "ca",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed4.neo.org",
    "location": "Canada",
    "address": "47.88.53.224",
    "locale": "ca",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed5.neo.org",
    "location": "Canada",
    "address": "47.91.92.192",
    "locale": "ca",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed1.cityofzion.io",
    "location": "USA",
    "locale": "us",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed2.cityofzion.io",
    "location": "USA",
    "locale": "us",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed3.cityofzion.io",
    "location": "USA",
    "locale": "us",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed4.cityofzion.io",
    "location": "USA",
    "locale": "us",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed5.cityofzion.io",
    "location": "USA",
    "locale": "us",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed0.cityofzion.io",
    "location": "CH",
    "locale": "ch",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed6.cityofzion.io",
    "location": "CH",
    "locale": "ch",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed7.cityofzion.io",
    "location": "AT",
    "locale": "at",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed8.cityofzion.io",
    "location": "AT",
    "locale": "at",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed9.cityofzion.io",
    "location": "AT",
    "locale": "at",
    "port": "443",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "node1.ams2.bridgeprotocol.io",
    "location": "Netherlands",
    "address": "198.211.127.188",
    "locale": "nl",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "node2.ams2.bridgeprotocol.io",
    "location": "Netherlands",
    "address": "37.139.22.119",
    "locale": "nl",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "node1.nyc3.bridgeprotocol.io",
    "location": "USA",
    "address": "165.227.186.72",
    "locale": "us",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "node2.nyc3.bridgeprotocol.io",
    "location": "USA",
    "address": "159.65.244.191",
    "locale": "us",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "node1.sgp1.bridgeprotocol.io",
    "location": "Singapore",
    "address": "206.189.94.221",
    "locale": "sg",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "node2.sgp1.bridgeprotocol.io",
    "location": "Singapore",
    "address": "206.189.152.158",
    "locale": "sg",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "pyrpc1.redpulse.com",
    "location": "Frankfurt",
    "locale": "de",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "pyrpc2.redpulse.com",
    "location": "Singapore",
    "locale": "sg",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed1.aphelion-neo.com",
    "location": "London",
    "address": "35.178.126.129",
    "locale": "gb",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed2.aphelion-neo.com",
    "location": "Frankfurt",
    "address": "18.197.136.206",
    "locale": "de",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed3.aphelion-neo.com",
    "location": "Seoul",
    "address": "52.78.136.233",
    "locale": "kr",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed4.aphelion-neo.com",
    "location": "Sao Paulo",
    "address": "54.233.224.215",
    "locale": "br",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "https",
    "url": "seed1.spotcoin.com",
    "location": "Frankfurt",
    "address": "18.194.110.81",
    "locale": "de",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed1.ngd.network",
    "location": "USA",
    "locale": "us",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed2.ngd.network",
    "location": "USA",
    "locale": "us",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed3.ngd.network",
    "location": "Hong Kong",
    "locale": "hk",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed4.ngd.network",
    "location": "Hong Kong",
    "locale": "hk",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed5.ngd.network",
    "location": "China",
    "locale": "cn",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed6.ngd.network",
    "location": "USA",
    "locale": "us",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed7.ngd.network",
    "location": "USA",
    "locale": "us",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed8.ngd.network",
    "location": "USA",
    "locale": "us",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed9.ngd.network",
    "location": "Hong Kong",
    "locale": "hk",
    "port": "10332",
    "type": "RPC"
  },
  {
    "protocol": "http",
    "url": "seed10.ngd.network",
    "location": "Hong Kong",
    "locale": "hk",
    "port": "10332",
    "type": "RPC"
  }
]
