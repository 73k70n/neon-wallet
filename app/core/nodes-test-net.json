[
  {
    "url": "seed1.switcheo.network",
    "protocol": "https",
    "port": "10331"
  },
  {
    "url": "seed3.switcheo.network",
    "protocol": "https",
    "port": "10331"
  },
  {
    "url": "seed1.travala.com",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed1.neo.org",
    "protocol": "https",
    "port": "10331"
  },
  {
    "url": "seed1.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed2.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed3.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed4.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed5.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed0.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed6.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed7.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed8.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed9.cityofzion.io",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed1.redpulse.com",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed2.redpulse.com",
    "protocol": "https",
    "port": "443"
  },
  {
    "url": "seed.o3node.org",
    "protocol": "https",
    "port": "10331"
  },
  {
    "url": "seed1.aphelion-neo.com",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed2.aphelion-neo.com",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed4.aphelion-neo.com",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed1.spotcoin.com",
    "protocol": "https",
    "port": "10332"
  },
  {
    "url": "rustylogic.ddns.net",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed1.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed2.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed3.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed4.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed5.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed6.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed7.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed8.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed9.ngd.network",
    "protocol": "http",
    "port": "10332"
  },
  {
    "url": "seed10.ngd.network",
    "protocol": "http",
    "port": "10332"
  }
]
