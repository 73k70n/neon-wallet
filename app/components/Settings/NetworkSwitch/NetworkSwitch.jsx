// @flow
import React, { Component } from 'react'

import { getNetworks } from '../../../core/networks'
import SelectInput from '../../Inputs/SelectInput/SelectInput'
import styles from './NetworkSwitch.scss'

type Props = {
  networkId: string,
  onChange: Function,
  networks: Array<NetworkItemType>
}

export default class NetworkSwitch extends Component<Props> {
  static defaultProps = {
    networks: getNetworks()
  }

  render() {
    const { networkId, networks } = this.props
    const { label } =
      (Array.isArray(networks) &&
        networks.find(network => network.id === networkId)) ||
      {}

    return (
      <div id="network" className={styles.networkSwitch}>
        <span className={styles.label}>Running on</span>
        <SelectInput
          className="NetworkSelector"
          items={
            Array.isArray(networks)
              ? networks.map(network => network.label)
              : []
          }
          value={label || ''}
          placeholder=""
          onChange={this.handleChange}
          getItemValue={value =>
            Array.isArray(networks) &&
            networks.find(network => network.label === value).id
          }
        />
      </div>
    )
  }

  handleChange = (value: string) => {
    this.props.onChange(value)
  }
}
