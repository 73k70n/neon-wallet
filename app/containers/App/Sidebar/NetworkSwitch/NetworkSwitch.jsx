// @flow
import React, { Component } from 'react'

import classNames from 'classnames'
import StyledReactSelect from '../../../../components/Inputs/StyledReactSelect/StyledReactSelect'
import { getNetworks } from '../../../../core/networks'
import styles from './NetworkSwitch.scss'

type Props = {
  onChange: Function,
  handleControlledChange: Function,
  className: string,
  disabled: boolean,
  networks: Array<NetworkItemType>,
  transparent: boolean,
  shouldSwitchNetworks: boolean,
  fontSize: number,
  value: NetworkItemType
}

// type State = {
//   option: NetworkItemType
// }

export default class NetworkSwitch extends Component<Props> {
  static defaultProps = {
    networks: getNetworks(),
    shouldSwitchNetworks: true
  }

  // state = {
  //   option:
  //     this.props.networks.find(
  //       network => network.id === this.props.networkId
  //     ) || this.props.networks[0]
  // }

  render() {
    const {
      networks,
      className,
      disabled,
      transparent,
      fontSize,
      value
    } = this.props

    return (
      <div id="network" className={classNames(styles.networkSwitch, className)}>
        <StyledReactSelect
          fontSize={fontSize}
          transparent={transparent}
          hideHighlight
          disabled={disabled}
          value={value}
          onChange={this.handleChange}
          options={networks}
          isSearchable={false}
        />
      </div>
    )
  }

  handleChange = (option: NetworkItemType) => {
    if (this.props.shouldSwitchNetworks) {
      return this.props.onChange(option.id)
    }
    this.props.handleControlledChange(option)
    // return this.setState({ option })
  }
}
