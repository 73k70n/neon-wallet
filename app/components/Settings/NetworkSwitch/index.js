// @flow
import { compose } from 'recompose'
import { withActions, type Actions } from 'spunky'

import NetworkSwitch from './NetworkSwitch'
import networkActions from '../../../actions/networkActions'
import withNetworkData from '../../../hocs/withNetworkData'

const mapActionsToProps = (actions: Actions): Object => ({
  onChange: networkId => actions.call({ networkId })
})

export default compose(
  withNetworkData(),
  withActions(networkActions, mapActionsToProps)
)(NetworkSwitch)
