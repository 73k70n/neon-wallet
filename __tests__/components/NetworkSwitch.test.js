import React from 'react'
import { shallow, mount } from 'enzyme'

import NetworkSwitch from '../../app/components/Settings/NetworkSwitch/NetworkSwitch'
import { MAIN_NETWORK_ID, TEST_NETWORK_ID } from '../../app/core/constants'

const MAIN_NET_LABEL = 'MainNet'

const setup = (shallowRender = true) => {
  const props = {
    networkId: MAIN_NETWORK_ID,
    networks: [
      {
        id: MAIN_NETWORK_ID,
        label: MAIN_NET_LABEL,
        network: 'MainNet'
      },
      {
        id: TEST_NETWORK_ID,
        label: 'TestNet',
        network: 'TestNet'
      }
    ],
    onChange: jest.fn()
  }

  let wrapper
  if (shallowRender) {
    wrapper = shallow(<NetworkSwitch {...props} />)
  } else {
    wrapper = mount(<NetworkSwitch {...props} />)
  }

  return {
    wrapper
  }
}

describe('NetworkSwitch', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('correctly renders MainNet initially', () => {
    const { wrapper } = setup()
    const networkSelectorElement = wrapper.find('.networkSelector').getElement();
    expect(networkSelectorElement.props.value).toEqual(MAIN_NET_LABEL)
  })

  test('switches to the correct network when chosen from the dropdown', () => {
    const { wrapper } = setup(false)
    const instance = wrapper.instance()

    const networkSelector = wrapper.find('.networkSelector input')

    networkSelector.simulate('change', { target: { value: TEST_NETWORK_ID } })
    expect(instance.props.onChange).toHaveBeenCalledWith(TEST_NETWORK_ID)

    networkSelector.simulate('change', { target: { value: MAIN_NETWORK_ID } })
    expect(instance.props.onChange).toHaveBeenCalledWith(MAIN_NETWORK_ID)
  })
})
