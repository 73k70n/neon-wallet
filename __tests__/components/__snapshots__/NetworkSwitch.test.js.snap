// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkSwitch renders without crashing 1`] = `
<div
  className="networkSwitch"
  id="network"
>
  <span
    className="label"
  >
    Network
  </span>
  <SelectInput
    className="networkSelector networkSelector"
    getItemValue={[Function]}
    getSearchResults={[Function]}
    items={
      Array [
        "MainNet",
        "TestNet",
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    placeholder=""
    renderAfter={[Function]}
    value="MainNet"
  />
</div>
`;
